File: doc-examples/jdbc-example-java/src/main/java/example/ContactView.java
package example;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.micronaut.data.annotation.GeneratedValue;
import io.micronaut.data.annotation.Id;
import io.micronaut.data.annotation.JsonView;
import io.micronaut.data.annotation.Relation;

import java.time.LocalDateTime;

// tag::record-example[]
@JsonView(value = "CONTACT_VIEW", alias = "cv", entity = Contact.class)
public record ContactView (
    @Id
    @GeneratedValue(GeneratedValue.Type.IDENTITY)
    Long id,
    String name,
    int age,
    LocalDateTime startDateTime,
    boolean active
) {}
// end::record-example[]

File: doc-examples/jdbc-example-python/src/main/python/example/ContactView.py
from dataclasses import dataclass
from typing import Annotated

from java.time import LocalDateTime
from micronaut.data.annotation import GeneratedValue, Id, JsonView

from example.Contact import Contact


# tag::record-example[]
@JsonView(value="CONTACT_VIEW", alias="cv", entity=Contact)
@dataclass
class ContactView:
    name: str
    age: int
    startDateTime: LocalDateTime | None = None
    active: bool = False
    id: Annotated[int | None, Id, GeneratedValue("IDENTITY")] = None
# end::record-example[]
